synapse/lib/cell.py

```diff
@@ -4031,3 +4047,219 @@ async def _runReloadFunc(self, name, func):
             return s_common.retnexc(e)
         logger.debug(f'Completed cell reload system {name}')
         return (True, ret)
+
+    async def addUserApiKey(self, useriden, name, duration=None):
+
+        user = await self.auth.reqUser(useriden)
+        iden, token, shadow = await s_passwd.generateApiKey()
+        now = s_common.now()
+        kdef = {
+            'iden': iden,
+            'name': name,
+            'user': user.iden,
+            'created': now,
+            'modified': now,
+            'shadow': shadow,
+            'duration': None,
+            'expref': None,
+        }
+
+        if duration:
```

Comment:
Any reason to not just compute now+duration and store a single `expires` key here?